src/DIRAC/WorkloadManagementSystem/DB/PilotAgentsDB.sql

```diff
@@ -65,3 +64,169 @@ CREATE TABLE `PilotOutput` (
   `StdError` MEDIUMTEXT,
   PRIMARY KEY (`PilotID`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
+
+-- ------------------------------------------------------------------------------
+-- summary tables and triggers
+-- ------------------------------------------------------------------------------
+
+-- distict values for GridType, GridSite, DestinationSite, Status and VO
+
+DROP TABLE IF EXISTS Summary_GridType;
+CREATE TABLE Summary_GridSite (
+  GridSite VARCHAR(128) PRIMARY KEY
+);
+DROP TABLE IF EXISTS Summary_DestinationSite;
+CREATE TABLE Summary_DestinationSite (
+  DestinationSite VARCHAR(128) PRIMARY KEY
+);
+DROP TABLE IF EXISTS Summary_Status;
+CREATE TABLE Summary_Status (
+  Status VARCHAR(128) PRIMARY KEY
+);
+DROP TABLE IF EXISTS Summary_VO;
+CREATE TABLE Summary_VO (
+  VO VARCHAR(128) PRIMARY KEY
+);
+
+DELIMITER //
+
+CREATE TRIGGER trg_pa_insert
+AFTER INSERT ON PilotAgents
+FOR EACH ROW
+BEGIN
+  INSERT IGNORE INTO Summary_GridSite (GridSite)
+  VALUES (NEW.GridSite);
+
+  INSERT IGNORE INTO Summary_DestinationSite (DestinationSite)
+  VALUES (NEW.DestinationSite);
+
+  INSERT IGNORE INTO Summary_Status (Status)
+  VALUES (NEW.Status);
+
+  INSERT IGNORE INTO Summary_VO (VO)
+  VALUES (NEW.VO);
+END;
+//
+
+DELIMITER ;
+
+DELIMITER //
+
+CREATE TRIGGER trg_pa_delete
+AFTER DELETE ON PilotAgents
+FOR EACH ROW
+BEGIN
+  IF NOT EXISTS (
+    SELECT 1 FROM PilotAgents WHERE GridType = OLD.GridType
```

Comment:
This looks like it has the potential to be expensive if we don't have an index on the column (in which case the trigger is pointless, the index does the same thing).

Reply:
This (GridType) is not needed. Are you also taking about the next lines (e.g. on GridSite?)

Reply:
I removed these tables now.